### PR TITLE
Feature/cdsct 195 add supplier tokens requests

### PR DESCRIPTION
**Pull request by cmarcus-nhs**
Look up the relevant token to use when making fhir/rest requests from inside the EMS.
Some things to note:
- Cactus services take precedence, followed by registered suppliers. If neither are present, no token is used.
- Still unsure of how to do any real higher level testing

### CODE REVIEW

File: src/main/java/uk/nhs/ctp/security/SupplierTokenResolver.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  private String fhirServer;
+
+  @Value("${blob.server}")
+  private String blobServer;
+
+  @Value("${ems.fhir.server}")
+  private String emsFhirServer;
+
+  @Value("${cactus.cdss}")
+  private String cactusCdss;
```
Comment (mhouston-nhs) [inline src/main/java/uk/nhs/ctp/security/SupplierTokenResolver.java:31]: I don't think this is a single baseURL now? We will need to update the registration result to account for multiple endpoints.

For this situation I would leave the CDSS out altogether and rely on providing the token, which would then allow us to test that the interceptor is working.

File: src/main/java/uk/nhs/ctp/service/CdssSupplierService.java [hunk 2]
```diff
@@ -61,6 +64,12 @@ public CdssSupplier getCdssSupplier(Long id) {
         .orElseThrow(EMSException::notFound);
   }
 
+  public Optional<CdssSupplier> findCdssSupplierByBaseUrl(String baseUrl) {
+    return cdssSupplierRepository.findAllBySupplierId(authService.requireSupplierId()).stream()
```
Comment (mhouston-nhs) [inline src/main/java/uk/nhs/ctp/service/CdssSupplierService.java:68]: As this is in the path of every request, I'd prefer to add an index on (supplierId, baseURL) and a query for looking up by those fields.

File: src/main/java/uk/nhs/ctp/service/EmsSupplierService.java [hunk 3]
```diff
@@ -21,6 +24,12 @@
         .findAllBySupplierId(authService.requireSupplierId());
   }
 
+  public Optional<EmsSupplier> findEmsSupplierByBaseUrl(String baseUrl) {
+    return emsSupplierRepository.findAllBySupplierId(authService.requireSupplierId()).stream()
```
Comment (mhouston-nhs) [inline src/main/java/uk/nhs/ctp/service/EmsSupplierService.java:28]: Also prefer using an index.

File: src/main/java/uk/nhs/ctp/service/CdssSupplierService.java (hunk 2 above)
Comment (cmarcus-nhs) [reply src/main/java/uk/nhs/ctp/service/CdssSupplierService.java:68]: Did consider that but didn't think there'd be many registered suppliers returned anyway. I'll add it

File: src/main/java/uk/nhs/ctp/security/SupplierTokenResolver.java (hunk 1 above)
Comment (cmarcus-nhs) [reply src/main/java/uk/nhs/ctp/security/SupplierTokenResolver.java:31]: Good point. We'll also need to add the 2.0 CDSS to the result of supplier registration at some point.

**Pull request merged**